apps/web/src/pages/JobPosterTemplateAdminPages/UpdateJobPosterTemplatePage/components/JobDetailsSection/JobDetailsSection.tsx

```diff
@@ -0,0 +1,285 @@
+
+export const InitialData_Fragment = graphql(/* GraphQL */ `
+  fragment UpdateJobPosterTemplateJobDetails on JobPosterTemplate {
+    id
+    name {
+      en
+      fr
+    }
+    description {
+      en
+      fr
+    }
+    supervisoryStatus {
+      value
+      label {
+        en
+        fr
+      }
+    }
+    workDescription {
+      en
+      fr
+    }
+    keywords {
+      en
+      fr
+    }
+    classification {
+      id
+      group
+      level
+    }
+    workStream {
+      id
+      name {
+        en
+        fr
+      }
+      community {
+        name {
+          en
+          fr
+        }
+      }
+    }
+    referenceId
+  }
+`);
+
+const Options_Fragment = graphql(/* GraphQL */ `
+  fragment UpdateJobPosterTemplateJobDetailsOptions on Query {
+    ...JobPosterTemplateJobDetailsFormOptions
+  }
+`);
+
+const UpdateJobPosterTemplateJobDetails_Mutation = graphql(/* GraphQL */ `
+  mutation UpdateJobPosterTemplateJobDetails(
+    $jobPosterTemplate: UpdateJobPosterTemplateInput!
+  ) {
+    updateJobPosterTemplate(jobPosterTemplate: $jobPosterTemplate) {
+      id
+    }
+  }
+`);
+
+const initialDataToFormValues = ({
+  name,
+  description,
+  supervisoryStatus,
+  workStream,
+  workDescription,
+  keywords,
+  classification,
+  referenceId,
+}: UpdateJobPosterTemplateJobDetailsFragment): FormValues => ({
+  jobTitleEn: name?.en ?? null,
+  jobTitleFr: name?.fr ?? null,
+  descriptionEn: description?.en ?? null,
+  descriptionFr: description?.fr ?? null,
+  supervisoryStatus: supervisoryStatus?.value ?? null,
+  workStreamId: workStream?.id ?? null,
+  workDescriptionEn: workDescription?.en ?? null,
+  workDescriptionFr: workDescription?.fr ?? null,
+  keywordsEn: (keywords?.en ?? []).join(", "),
+  keywordsFr: (keywords?.fr ?? []).join(", "),
+  classificationGroup: classification?.group ?? null,
+  classificationLevel: classification?.id ?? null,
+  referenceId: referenceId ?? null,
+});
+
+const formValuesToMutationInput = (
+  id: Scalars["UUID"]["input"],
+  {
+    jobTitleEn,
+    jobTitleFr,
+    descriptionEn,
+    descriptionFr,
+    supervisoryStatus,
+    workStreamId,
+    workDescriptionEn,
+    workDescriptionFr,
+    keywordsEn,
+    keywordsFr,
+    classificationLevel,
+    referenceId,
+  }: FormValues,
+): UpdateJobPosterTemplateInput => ({
+  id: id,
+  name: {
+    en: jobTitleEn,
+    fr: jobTitleFr,
+  },
+  description: {
+    en: descriptionEn,
+    fr: descriptionFr,
+  },
+  supervisoryStatus: supervisoryStatus as SupervisoryStatus,
+  workStream: {
+    connect: workStreamId,
+  },
+  workDescription: {
+    en: workDescriptionEn,
+    fr: workDescriptionFr,
+  },
+  keywords: {
+    en: keywordsEn?.split(",").map((s) => s.trim()),
+    fr: keywordsFr?.split(",").map((s) => s.trim()),
+  },
+  classification: {
+    connect: classificationLevel, // the ID for the group-level is in the level input
+  },
+  referenceId: referenceId,
+});
+
```

Comment:
Not actionable since it is purely subjective and this is on a deadline but, you can use dot syntax in the forms to simplify this and `react-hook-form` will interpret that as an object.

```tsx
type FormValues {
  jobTtile: {
    en: string;
    fr: string;
  }
};

<Input name="jobTitle.en" />
<Input name="jobTitle.fr" />

const formValuesToMutationInput = (values) => ({
    name: values.jobTitle,
})
```

Reply:
Yeah, it's pretty cool.  I've just found that unless you can use the exact same type for your FormValues as your mutation input, it doesn't really save any time since you have to map to a new object anyway.  :laughing: Is there another benefit I haven't thought of?